.vsts-dotnet.yml

```
@@ -10,12 +10,28 @@ trigger:
 #   SignType: real
 #   SkipApplyOptimizationData: false
 
+parameters:
+- name: OptProfDropName
+  displayName: Optional OptProfDrop Override
+  type: string
+  default: 'default'
+
 variables:
+  # if OptProfDrop is not set, string '$(OptProfDrop)' will be passed to the build script.
+  - name: OptProfDrop
+    value: ''
   - name: SourceBranch
     value: $(IbcSourceBranchName)
-  - ${{ if startsWith(variables['Build.SourceBranch'], 'refs/heads/exp/') }}:
+  # If we're not on a vs* branch, use main as our optprof collection branch
+  - ${{ if not(startsWith(variables['Build.SourceBranch'], 'refs/heads/vs')) }}:
     - name: SourceBranch
       value: main
+  # if OptProfDropName is set as a parameter, set OptProfDrop to the parameter and unset SourceBranch
+  - ${{ if ne(parameters.OptProfDropName, 'default') }}:
+    - name: OptProfDrop
+      value: ${{parameters.OptProfDropName}}
+    - name: SourceBranch
+      value: ''
```

I know pretty much nothing about yml. Curious, why this can't be done with `OptProfDropName` alone, i.e. without introducing `OptProfDrop`. Assuming that `''` is the default value of the parameter, can we do the following and omit the rest of the changes except for the `refs/heads/exp/` -> !`refs/heads/vs` tweak?

```suggestion
  - ${{ if ne(parameters.OptProfDropName, '') }}:
    - name: SourceBranch
      value: ''
```

---

The main thing to solve with `OptProfDropName` is making sure that further in the script `p:VisualStudioIbcDrop` is passed an empty string if we don't care to override it. Because `OptProfDropName` is a pipeline parameter now, it **can't** have an empty value 😞 (maybe it can be set to empty during the script compile time?). 

So I have that `OptProfDrop` variable because:
1. I know how to work with `OptProfDrop`as a variable at this point (ie. I know I can set it equal to an empty string)
2. It took ~28 commits to get to this point and I _really_ don't want to keep testing to get a cleaner yml for minimal benefit

---

Makes perfect sense, thank you!

---

Note that this isn't so much about "YAML" as about "Azure DevOps Pipelines YAML" which is similar to "GitHub Actions YAML" but (maybe?) not exactly the same.

Having been in Ben's position I completely empathize with

> It took ~28 commits to get to this point and I _really_ don't want to keep testing to get a cleaner yml for minimal benefit

!